### PR TITLE
V3 - add warning when customers registers HttpClient using DI 

### PR DESCRIPTION
**Pull request by madelinegordon**
### Issue describing the changes in this PR

resolves part of #7585. Still need to implement v4 breaking change. 

### Pull request checklist

* [ ] My changes **do not** require documentation changes
    * [x] Otherwise: Documentation issue linked to PR
* [ ] My changes **should not** be added to the release notes for the next release
    * [x] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [ ] I have added all required tests (Unit tests, E2E tests)

### Additional information

Non-breaking change for V3.

### CODE REVIEW

File: src/WebJobs.Script.WebHost/WebScriptHostBuilderExtension.cs [hunk 1]
```diff
@@ -54,6 +55,13 @@ public static class WebScriptHostBuilderExtension
                 {
                     configureWebJobs?.Invoke(webJobsBuilder);
 
+                    webJobsBuilder.Services.TryAddSingleton<HttpClient>(f =>
+                    {
+                        var loggerFactory = f.GetService<ILoggerFactory>();
+                        loggerFactory.CreateLogger("Host.Startup").LogWarning("Detected direct usage of HttpClient in function app. Please use HttpClientFactory instead - see <TODO aka.ms link> for more information.");
```
Comment (brettsam) [inline src/WebJobs.Script.WebHost/WebScriptHostBuilderExtension.cs]: Suggesting an updated message: `Using HttpClient as an injected dependency will not be supported in future versions of Azure Functions. Use IHttpClientFactory instead. See http://aka.ms/functions-httpclient-di for more information.`

**Review by liliankasem — DISMISSED** [review]

File: src/WebJobs.Script.WebHost/WebScriptHostBuilderExtension.cs [hunk 2]
```diff
@@ -54,6 +55,13 @@ public static class WebScriptHostBuilderExtension
                 {
                     configureWebJobs?.Invoke(webJobsBuilder);
 
+                    webJobsBuilder.Services.TryAddSingleton<HttpClient>(f =>
+                    {
+                        var loggerFactory = f.GetService<ILoggerFactory>();
+                        loggerFactory.CreateLogger("Host.Startup").LogWarning("Using HttpClient as an injected dependency will not be supported in future versions of Azure Functions. Use IHttpClientFactory instead. See http://aka.ms/functions-httpclient-di for more information.");
```
Comment (brettsam) [inline src/WebJobs.Script.WebHost/WebScriptHostBuilderExtension.cs]: nit: can you use `LogCategories.Startup` constant instead?

**Review by brettsam — COMMENTED** [review]
Just one tiny comment!

File: release_notes.md [hunk 3]
```diff
@@ -2,9 +2,10 @@
 <!-- Please add your release notes in the following format:
 - My change description (#PR)
 -->
-- My change description (#PR)
 
+- Added warning for customer registering HttpClient with DI (#7674)
 - Updated Java Worker Version to [1.9.0](https://github.com/Azure/azure-functions-java-worker/releases/tag/1.9.0)
 
+
```
Comment (liliankasem) [inline release_notes.md:9]: super nit: accidental extra line?

**Review by liliankasem — APPROVED** [review]

**Pull request merged**